src/WhatToCook.Application/Infrastructure/Repositories/RecipesRepository.cs

```diff
@@ -7,45 +7,72 @@ namespace WhatToCook.Application.Infrastructure.Repositories;
 
 public interface IRecipesRepository
 {
-    Task<Recipe?> GetRecipeByName(string name);
-    List<Recipe> GetRecipesByNameForMealPlan(IEnumerable<string> names);
     Task Create(Recipe recipe);
-    Task Update(Recipe recipe);
-    Task<string> SaveImage(ImageInfo imageInfo);
+
     Task Delete(int id);
+
+    Task<Recipe?> GetRecipeByName(string name);
+
+    List<Recipe> GetRecipesByIdForMealPlan(IEnumerable<int> ids);
+
+    Task<string> SaveImage(ImageInfo imageInfo);
+
+    Task Update(Recipe recipe);
 }
 
 public class RecipesRepository : IRecipesRepository
 {
     private readonly DatabaseContext _dbContext;
-    private readonly ILogger _logger;
     private readonly IFileSaver _fileSaver;
+    private readonly ILogger _logger;
+
     public RecipesRepository(DatabaseContext dbContext, ILogger<RecipesRepository> logger, IFileSaver fileSaver)
     {
         _dbContext = dbContext;
         _logger = logger;
         _fileSaver = fileSaver;
     }
 
-    public List<Recipe> GetRecipesByNameForMealPlan(IEnumerable<string> names)
+    public async Task Create(Recipe recipe)
     {
-        var recipes = _dbContext.Recipes.Include(recipe => recipe.PlansOfMeals)
-        .Where(recipe => names.Contains(recipe.Name)).ToList();
-        var existingRecipeNames = recipes.Select(r => r.Name).ToList();
-        if (!existingRecipeNames.OrderBy(n => n).SequenceEqual(names.OrderBy(n => n)))
-        {
-            var missingRecipeNames = existingRecipeNames.Except(names);
-            var errorMessage = $"Recipes do not exist in the database: {string.Join(", ", missingRecipeNames)}";
+        await _dbContext.Recipes.AddAsync(recipe);
+        await _dbContext.SaveChangesAsync();
+    }
 
-            _logger.LogError(errorMessage); 
-            throw new Exception(errorMessage);
+    public async Task Delete(int id)
+    {
+        var recipe = await _dbContext.Recipes.FindAsync(id);
```

Comment:
`var recipe = await _dbContext.Recipes.FindAsync(id);`
Can be combined with null check and exception throw in this way:

`var recipe = await _dbContext.Recipes.FindAsync(id) ?? throw new NotFoundException($"Recipe with ID '{id}' not found.")`

So if recipe will be null it will throw exception. Code will be a little simpler and if that checks null can be removed